### PR TITLE
feat(lambda-event-sources): events source mapping support for sqs max concurrency

### PR DESCRIPTION
**Pull request by dazzle1979**
This pull request adds support for maximum concurrency for Amazon SQS event sources: https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#events-sqs-max-concurrency

----

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing guide?](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md)

### Adding new Construct Runtime Dependencies:

~~* [ ] This PR adds new construct runtime dependencies following the process described [here](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md/#adding-construct-runtime-dependencies)~~

### New Features

* [x] Have you added the new feature to an [integration test](https://github.com/aws/aws-cdk/blob/main/INTEGRATION_TESTS.md)?
	* [x] Did you use `yarn integ` to deploy the infrastructure and generate the snapshot (i.e. `yarn integ` without `--dry-run`)?

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by TheRealAmazonKendra — DISMISSED** [review]
Thank you for your contribution! Overall this looks pretty fantastic. I just have a few comments inline.

File: packages/@aws-cdk/aws-lambda-event-sources/test/integ.sqs-max-concurrency.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import * as sqs from '@aws-cdk/aws-sqs';
```
[the remark this answers is not in the document]
Comment (dazzle1979) [reply packages/@aws-cdk/aws-lambda-event-sources/test/integ.sqs-max-concurrency.ts:1]: Updated the test, using the IntegTest construct now

File: packages/@aws-cdk/aws-lambda-event-sources/test/sqs.test.ts [hunk 2]
```diff
@@ -423,4 +423,28 @@ describe('SQSEventSource', () => {
       },
     });
   });
+
+  test('fails if maxConcurrency < 2', () => {
```
[the remark this answers is not in the document]
Comment (dazzle1979) [reply packages/@aws-cdk/aws-lambda-event-sources/test/sqs.test.ts:427]: done!

File: packages/@aws-cdk/aws-lambda/lib/event-source-mapping.ts [hunk 3]
```diff
@@ -372,8 +392,9 @@ export class EventSourceMapping extends cdk.Resource implements IEventSourceMapp
       parallelizationFactor: props.parallelizationFactor,
       topics: props.kafkaTopic !== undefined ? [props.kafkaTopic] : undefined,
       tumblingWindowInSeconds: props.tumblingWindow?.toSeconds(),
-      sourceAccessConfigurations: props.sourceAccessConfigurations?.map((o) => {return { type: o.type.type, uri: o.uri };}),
+      scalingConfig,
       selfManagedEventSource,
+      sourceAccessConfigurations: props.sourceAccessConfigurations?.map((o) => {return { type: o.type.type, uri: o.uri };}),
```
[the remark this answers is not in the document]
Comment (dazzle1979) [reply packages/@aws-cdk/aws-lambda/lib/event-source-mapping.ts:391]: I moved it back to its original location. I like to have things in alphabetic order, but moving this line does not fix that issue for this file  :)

File: packages/@aws-cdk/aws-lambda/lib/event-source-mapping.ts [hunk 4]
```diff
@@ -354,6 +369,11 @@ export class EventSourceMapping extends cdk.Resource implements IEventSourceMapp
       selfManagedEventSource = { endpoints: { kafkaBootstrapServers: props.kafkaBootstrapServers } };
     }
 
+    let scalingConfig;
+    if (props.maxConcurrency) {
+      scalingConfig = { maximumConcurrency: props.maxConcurrency };
+    }
```
[the remark this answers is not in the document]
Comment (dazzle1979) [reply packages/@aws-cdk/aws-lambda/lib/event-source-mapping.ts]: By doing it that way, we will include the scalingConfig prop always and everyone needs to update their snapshots. I'm now doing it a bit differently (cleaner) and scalingConfig will only be added when maxConcurrency is set.

**Comment by dazzle1979** [comment]
Thanks for your review 🙏 I have addressed all your comments, and hope it is ok now.

**Review by TheRealAmazonKendra — DISMISSED** [review]

**Review by TheRealAmazonKendra — APPROVED** [review]

**Pull request merged**